typing/typecore.ml

```diff
@@ -3684,11 +3686,10 @@ and type_expect_
       ruem ~mode ~expected_mode {
         exp_desc; exp_loc = loc; exp_extra = [];
         exp_type = desc.val_type;
-        exp_mode = expected_mode.mode;
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
   | Pexp_constant(Pconst_string (str, _, _) as cst) ->
-    register_allocation expected_mode;
+    ignore (register_allocation expected_mode);
```

Comment:
This `ignore` suggests that this call to `register_allocation` probably isn't needed.

Reply:
I assume so, since there isn't an associated alloc mode in the typedtree.